src/mongoc/mongoc-database.c

```diff
@@ -618,6 +618,7 @@ mongoc_database_has_collection (mongoc_database_t *database,
        (error &&
         ((error->domain != 0) ||
          (error->code != 0)))) {
+      mongoc_cursor_destroy( cursor );
```

Comment:
If we enter this clause with !cursor then mongoc_cursor_destroy() is going to be very pissed off.